app/code/Magento/MediaStorage/Model/File/Storage/File.php

```diff
@@ -291,16 +291,15 @@ public function saveFile($file, $overwrite = true)
             && isset($file['content'])
             && !empty($file['content'])
         ) {
-            try {
-                $filename = isset(
-                    $file['directory']
-                ) && !empty($file['directory']) ? $file['directory'] . '/' . $file['filename'] : $file['filename'];
+            $directory = !empty($file['directory'] ?? '') ? $file['directory'] . '/' : '';
+            $filename = $directory . $file['filename'];
 
+            try {
                 return $this->_fileUtility->saveFile($filename, $file['content'], $overwrite);
             } catch (\Exception $e) {
                 $this->_logger->critical($e);
                 throw new \Magento\Framework\Exception\LocalizedException(
-                    __('Unable to save file "%1" at "%2"', $file['filename'], $file['directory'])
+                    __('Unable to save file "%1" at "%2"', $file['filename'], $filename)
```

Comment:
Maybe, now we should pass $directory here?
```suggestion
                    __('Unable to save file "%1" at "%2"', $file['filename'], $directory)
```

Reply:
Ah yes. I didn't want to change too much of the original workings, but it actually makes more sense to make it clear no directory is present for example? Otherwise the message would be "Unable to save file filename.ext at " that would look weird in my opinion.

Reply:
ok, let's keep it as is.